docs/docs/dev-plugins.mdx

```diff
@@ -0,0 +1,120 @@
+---
+id: dev-plugins
+title: Plugin development
+---
+
+Vitessce supports plugin view types, coordination types, and file types.
+
+Plugin view types and coordination types can enable development of interactive visualizations to support custom use cases (i.e., use cases beyond those supported by existing view types).
+Plugin file types can enable loading data that is stored in custom file formats.
+
+:::caution
+
+Note that we cannot guarantee that future versions of Vitessce will not break existing plugins.
+For instance, we do not make any guarantees that exported helper functions such as React data-loading hooks (e.g., `useRasterData`, `useExpressionMatrixData`, etc.) will not be removed or will have a consistent function signature across Vitessce versions.
```

Comment:
I'm hesitant not to at least say "we will make an attempt to make breaking changes only on minor version updates" - this seems like a low bar to clear, but it's more labor for us to be careful a bout version updates.

Reply:
Done in https://github.com/vitessce/vitessce/pull/1212/commits/2bb50629f379b1fc44869c4eea56665eaea80ce8